lib/buffer.js

```diff
@@ -503,12 +503,12 @@ Buffer.prototype.copy = function(target, targetStart, sourceStart, sourceEnd) {
   return binding.copy(this, target, targetStart, sourceStart, sourceEnd);
 };
 
-Buffer.prototype.toString = function() {
+Buffer.prototype.toString = function(encoding, start, end) {
   let result;
   if (arguments.length === 0) {
     result = this.utf8Slice(0, this.length);
   } else {
-    result = slowToString.apply(this, arguments);
+    result = slowToString.call(this, encoding, start, end);
```

Comment:
Will require a bit of editing inside `slowToString` which assumes it is bound to `this`, but that's certainly doable.